### PR TITLE
Update move sec docs

### PR DESCRIPTION
**Pull request by GotenJBZ**
### Description
Introducing:
1. Object Ownership Check
2. randomness - undergasing
3. randomness - test-and-abort
### Checklist

- Do all Lints pass?
  - [ ] Have you ran `pnpm spellcheck`?
  - [x] Have you ran `pnpm fmt`?
  - [x] Have you ran `pnpm lint`?

### CODE REVIEW

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (gedigi) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:37]: it's not clear what this code does or why it matters that the ownership of the object is checked.

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (gedigi) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:34]: shouldn't this be `object::transfer()`?

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (gedigi) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:657]: isn't this prevented at compile time?

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (gedigi) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:739]: from the example it's not clear why they consume different amounts of gas.

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (gedigi) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:750]: should we add a link to the complete randomness documentation?

Comment (gregnazario) [reply apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:750]: +1 please add appropriate randomness docs link here

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (gregnazario) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:740]:
```suggestion
If the `lose` function consumes more gas than the `win` function, an attacker can set the max gas limit that is sufficient for `win` but not for `lose`. This forces the transaction to abort when the `lose` path is taken, ensuring that the user will never execute the `lose` path.  Then, the user can call the function repeatedly until they win.
```

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (gregnazario) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:714]: This is repeated in the paragraph below, let's clean this up
```suggestion
When different code paths in a function consume different amounts of gas, an attacker can manipulate the gas limit to bias the outcome.  Let's look at an example of how different paths can consume different amounts of gas.
```

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (gregnazario) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:673]: Can we stay concrete and the same for these examples?

Here we use WIN, and below we use Object_for_the_winner
```suggestion
```move
module user::lottery {
    struct Winner {}

    fun mint_to_user(user: &signer) {
        move_to(address_of(user), Winner{});
    }

    #[randomness]
    public entry fun play(user: &signer) {
        let random_value = aptos_framework::randomness::u64_range(0, 100);
        if (random_value == 42) {
            mint_to_user(user);
        }
    }
```

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (gregnazario) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:18]: This isn't really any different than a resource as well?  Maybe we should mention that

**Review by gregnazario — COMMENTED** [review]
Let's please make these updates to Nextra as well, we won't be migrating the changes

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (zi0Black) [reply apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:657]: Yes, https://github.com/aptos-labs/aptos-core/pull/13294

Comment (zi0Black) [reply apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:657]: May we nonetheless document this as an invariant?

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (GotenJBZ) [reply apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:780]: Added at the beginning of the randomness section

**Review by hariria — COMMENTED** [review]
@GotenJBZ What's the status on this? If it's mostly complete can we merge it and add changes as a fast follow? I'm restructuring the docs in smart contracts so wanted to avoid merge conflict nightmares later if possible.

File: apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md
Comment (hariria) [inline apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md:735]:
```suggestion
```move filename="randomness_example.move"
```

File: apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx
Comment (hariria) [inline apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx:57]:
```suggestion
```move filename="object_example.move"
```

File: apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx
Comment (hariria) [inline apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx:22]:
```suggestion
```move filename="object_example.move"
```

File: apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx
Comment (hariria) [inline apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx:732]:
```suggestion
```move filename="randomness_example.move"
```

File: apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx
Comment (hariria) [inline apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx:750]:
```suggestion
```move filename="randomness_example.move"
```

File: apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx
Comment (hariria) [inline apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx:763]:
```suggestion
```move filename="randomness_example.move"
```

File: apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx
Comment (hariria) [inline apps/nextra/pages/en/build/smart-contracts/learn-move/advanced-guides/move-security-guidelines.mdx:785]:
```suggestion
```move filename="randomness_example.move"
```

**Review by hariria — APPROVED** [review]
Some explicit titles for the Move code snippets on Nextra will help folks i think

**Review by gedigi — APPROVED** [review]

**Comment by GotenJBZ** [comment]
> @GotenJBZ What's the status on this? If it's mostly complete can we merge it and add changes as a fast follow? I'm restructuring the docs in smart contracts so wanted to avoid merge conflict nightmares later if possible.

It's done, I was waiting for approvals. By the way I also add the content to nextra

**Comment by GotenJBZ** [comment]
> Some explicit titles for the Move code snippets on Nextra will help folks i think

I was not aware that this was possible. After the merge I will make another pr adding the title

**Comment by hariria** [comment]
I'll go ahead and commit the suggestions above and merge

**Comment by GotenJBZ** [comment]
> I'll go ahead and commit the suggestions above and merge

maybe it's better `randomness_secure_example.move` and `randomness_insecure_example.move`

**Pull request merged**
